lib/uplinkc/convert.go

```diff
@@ -14,30 +14,49 @@ func newBucketInfo(bucket *storj.Bucket) C.BucketInfo {
 	return C.BucketInfo{
 		name:         C.CString(bucket.Name),
 		created:      C.int64_t(bucket.Created.Unix()),
-		path_cipher:  C.uint8_t(bucket.PathCipher),
+		path_cipher:  toCCipherSuite(bucket.PathCipher),
 		segment_size: C.uint64_t(bucket.SegmentsSize),
 
 		encryption_parameters: convertEncryptionParameters(&bucket.EncryptionParameters),
 		redundancy_scheme:     convertRedundancyScheme(&bucket.RedundancyScheme),
 	}
 }
 
+// newObjectInfo returns a C object struct converted from a go object struct.
+func newObjectInfo(object *storj.Object) C.ObjectInfo {
+	return C.ObjectInfo{
+		version:      C.uint32_t(object.Version),
+		bucket:       newBucketInfo(&object.Bucket),
+		path:         C.CString(object.Path),
+		is_prefix:    C.bool(object.IsPrefix),
+		content_type: C.CString(object.ContentType),
+		created:      C.int64_t(object.Created.Unix()),
+		modified:     C.int64_t(object.Modified.Unix()),
+		expires:      C.int64_t(object.Expires.Unix()),
+	}
+}
+
 // convertEncryptionParameters converts Go EncryptionParameters to C.
 func convertEncryptionParameters(goParams *storj.EncryptionParameters) C.EncryptionParameters {
 	return C.EncryptionParameters{
-		cipher_suite: C.uint8_t(goParams.CipherSuite),
+		cipher_suite: C.CipherSuite(goParams.CipherSuite),
```

Comment:
use a func here as well, so I can sleep nicer today :)

Reply:
is this a nitpick?

Reply:
I'm not sure what you're asking for here. The go type is `CipherSuite`. This is the correct cast from the go to the C respective types, we're not converting between cipher and cipher suite here and there's no way to know the difference.